notifications/v1/types.go

```diff
@@ -0,0 +1,16 @@
+package v1
+
+// ⚠️ WARNING: These types MUST match the SAMS implementation, at
+// backend/internal/notification/types.go
+
+const (
+	nameUserDeleted = "UserDeleted"
+)
+
+// UserDeletedData contains information of a "UserDeleted" notification.
+type UserDeletedData struct {
+	// UserID is the SAMS external user ID of the deleted user.
+	UserID string `json:"user_id"`
```

Comment:
`user_id` isn't really an idiomatic name for a JSON field. Is there a particular reason we are using that instead of `userId`? Similarly, does this actually matter as long as SAMS and the SAMS SDK are in agreement (e.g. it could be `xxxasdfkjl2yolo`)?

Reply:
> `user_id` isn't really an idiomatic name for a JSON field.

Really? 😂 For all my life I have believed that's the convention (maybe I used too much GitHub API?), and all the OAuth/OIDC fields uses snake case 🤔

Reply:
OK, perhaps you are right and I'm speaking nonsense. (Sorry.) Apparently my suggestion is only 0.33333 reasonable 😭 .

1. If the RFC or some spec written by people much smarter than me does it some way, we should do it that way too. (So ❌ for Chris.)
2. Anything GitHub does is perfect. Therefore, if that is how they name identifiers in their API, it is how things should be done. (Though I'd guess it has something to do with the service's roots in Ruby...) (So ❌ for Chris.)
3. JSON objects are used at runtime in JavaScript, and therefore would be subject to the Google JavaScript Style Guide. (So 🟢 for Chris.)

https://google.github.io/styleguide/jsguide.html#naming-non-constant-field-names

Reply:
I always associated fooBar with JSON and foo_bar with YAML, to throw some noise into the debate :)

Reply:
I do notice convention is JS world and non-JS is very different, my understanding is that's simply because by using camel case in JS, that JSON field can be referenced directly as a field of an object (`T.userId` ✅ ), but not so if using snake case (`T.user_id` ❌ ;GraphQL counts as JS world too).

However, outside JS world, I've only found snake case to have conventional look and feel.